nucypher/characters/base.py

```diff
@@ -113,7 +116,20 @@ def __init__(self,
 
         #
         # Operating Mode
-        #
+
+        if is_me:
+            if not node_class:
+                # Once in a while, in tests or demos, we init a plain Character who doesn't already know about its node class.
+                from nucypher.characters.lawful import Ursula
+                node_class = Ursula
+            # If we're federated only, we assume that all other nodes in our domain are as well.
+            node_class.set_federated_mode(federated_only)
+        else:
+            # What an awful hack.  The last convulsions of #466.
```

Comment:
What's the alternative? Is there an issue for this?

Reply:
As far as I'm concerned, #466 *is* the issue.  Do you think there's a need for a distinct one?